### PR TITLE
MTDSA-3203 Updated models and tests

### PR DESCRIPTION
**Pull request by narendravijayarao**

### CODE REVIEW

File: app/v2/models/EoyCharitableGiving.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+package v2.models
+
+import play.api.libs.functional.syntax._
+import play.api.libs.json.Reads._
+import play.api.libs.json.{Json, _}
+
+case class EoyCharitableGiving(taxableIncome: BigDecimal,
+                               supplied: Boolean,
+                               finalised: Option[Boolean])
```
Comment (jeremystone) [inline app/v2/models/EoyCharitableGiving.scala]: Does charitable giving have a finalised field? Does not appear to be in spec...?

File: app/v2/models/EoySavings.scala [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+package v2.models
+
+import play.api.libs.functional.syntax._
+import play.api.libs.json.Reads._
+import play.api.libs.json.{Json, _}
+
+case class EoySavings(savingsAccountId: String,
+                      taxableIncome: BigDecimal,
+                      supplied: Boolean,
+                      finalised: Option[Boolean])
```
Comment (jeremystone) [inline app/v2/models/EoySavings.scala]: Can't see this finalised in spec either?

File: app/v2/services/TaxCalcService.scala [hunk 3]
```diff
@@ -30,6 +30,8 @@ import scala.concurrent.{ExecutionContext, Future}
 class TaxCalcService @Inject()(connector: TaxCalcConnector) {
 
   private val calcIdRegex = "^[0-9]{8}$|^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$"
+  private val savingsAccountIdRegex = "^[A-Za-z0-9]{15}$"
```
Comment (jeremystone) [inline app/v2/services/TaxCalcService.scala]: Does not appear to be used

Comment (narendravijayarao) [reply app/v2/services/TaxCalcService.scala]: Not sure about this... Already there, if this is not used anywhere we can remove.

File: app/v2/models/EoySavings.scala (hunk 2 above)
Comment (narendravijayarao) [reply app/v2/models/EoySavings.scala]: I forgot, need to check with Scott

**Pull request merged**